adafruit_simplemath.py

```diff
@@ -27,12 +28,72 @@ def map_range(
     x: float, in_min: float, in_max: float, out_min: float, out_max: float
 ) -> float:
     """
-    Maps a number from one range to another. Somewhat similar to the Arduino ``map()`` function,
-    but returns a floating point result, and constrains the output value to be between
-    ``out_min`` and ``out_max``.
-    If ``in_min`` is greater than ``in_max`` or ``out_min`` is greater than ``out_max``,
-    the corresponding range is reversed, allowing, for example, mapping a range of 0-10 to 50-0.
+    Maps a number from one range to another. Somewhat similar to the Arduino
+    :attr:`map()` function, but returns a floating point result, and
+    constrains the output value to be between :attr:`out_min` and
+    :attr:`out_max`. If :attr:`in_min` is greater than :attr:`in_max` or
+    :attr:`out_min` is greater than :attr:`out_max`, the corresponding range
+    is reversed, allowing, for example, mapping a range of 0-10 to 50-0.
 
+    See also :py:func:`unconstrained_map_range`
+
+    .. code-block::
+
+        from adafruit_simplemath import map_range
+
+        percent = 23
+        screen_width = 320  # or board.DISPLAY.width
+        x = map_range(percent, 0, 100, 0, screen_width - 1)
+        print("X position", percent, "% from the left of screen is", x)
+
+    :param float x: Value to convert
+    :param float in_min: Start value of input range.
+    :param float in_max: End value of input range.
+    :param float out_min: Start value of output range.
+    :param float out_max: End value of output range.
+    :return: Returns value mapped to new range.
+    :rtype: float
+    """
+    # in_range = in_max - in_min
+    # in_delta = x - in_min
+    # if in_range != 0:
+    #     mapped = in_delta / in_range
+    # elif in_delta != 0:
+    #     mapped = in_delta
+    # else:
+    #     mapped = 0.5
+    # mapped *= out_max - out_min
+    # mapped += out_min
+
+    mapped = unconstrained_map_range(x, in_min, in_max, out_min, out_max)
+
+    if out_min <= out_max:
+        return max(min(mapped, out_max), out_min)
+    return min(max(mapped, out_max), out_min)
```

Comment:
You could replace this with a call to `constrain()`, right?

Reply:
Yes. I'd somehow convinced by brain that the auto-swapping in constrain() would break this but it doesn't.

I've replaced with a call to constrain() and added some tests for out-of-range descending outputs.